### PR TITLE
Composer size when re-entering with unfinished multiline message

### LINKED ISSUE #17361: Composer: can't see the entire multiline message in collapsed composer after reopening the chat [Android]
# Bug Report

## Problem 

https://github.com/status-im/status-mobile/assets/67952253/fd897a1a-edfd-4c73-8f77-a6e21e345b32

#### Steps:
1. Open chat
2. Type a multiline message
3. Close the chat and open it again
4. Try to set the cursor to the end of the message

### Additional Information

- Status version: nightly 20/09
- Operating System: Android

### PR DESCRIPTION
**Pull request by clauxx**
fixes #17361 

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)

When re-entering the composer while there's an unfinished multiline message, the composer wouldn't show it properly. Even when focused, it only showed one line. Now, in the same scenario, it will show the first two lines of the message when not focused and the entire message when focused, as usual.

NOTE: additionally, added a fix to the composer bottom shadow, which should be shown when the composer is collapsed and has a multiline message.

#### Platforms

- Android
- iOS

##### Functional

- 1-1 chats
- public chats
- group chats

status: ready

### CODE REVIEW

**Review by mohsen-ghafouri — APPROVED** [review]
👍🏻

**Review by OmarBasem — APPROVED** [review]

**Comment by churik** [comment]
@clauxx 
there is a new issue related to reopening chat.

### ISSUE 1: The composer jumps off the screen is reopened chat with a long message 

**Steps:**
1) paste a long message (to expand composer to the max height)
2) reopen channel
3) focus on the composer


**Actual result:**

https://github.com/status-im/status-mobile/assets/4557972/af2c780f-b77d-4fae-8923-f6b1dc4a8c51


**OS:** IOS, Android

**Review by FFFra — APPROVED** [review]

**Comment by clauxx** [comment]
Hey @churik, sorry for the long delay on this issue, was out sick last week. I fixed the issue you mentioned, but noticed that when re-entering the screen when there's an image added to the composer it focuses when opening the chat. This is different from the usual behavior, where it doesn't focus automatically. Is it intended?

Also, noticed the composer bottom shadow was not visible for multiline messages when it's collapsed, so fixed that as well.

**Comment by churik** [comment]
@clauxx the behavior is definitely better, but still see that the composer jumps of and part of the text is hidden behind the keyboard until you start typing. The issue is reproducible on IOS only.

https://github.com/status-im/status-mobile/assets/4557972/15173df3-ab9f-49b5-8638-007b240bce85

**Comment by flexsurfer** [comment]
hey @clauxx any updates? still relevant?

**Comment by clauxx** [comment]
@flexsurfer just came back from vacation. Will look check it out tomorrow.

File: src/status_im/contexts/chat/composer/gradients/style.cljs [hunk 1]
```diff
@@ -31,13 +31,14 @@
...
 (defn bottom-gradient
   []
-  {:colors [(colors/theme-colors colors/white colors/neutral-95)
-            (colors/theme-colors colors/white-opa-0 colors/neutral-95-opa-0)]
-   :start  {:x 0 :y 1}
-   :end    {:x 0 :y 0}
-   :style  bottom-gradient-style})
+  {:colors    [(colors/theme-colors colors/white-opa-90 colors/neutral-95-opa-60)
+               (colors/theme-colors colors/white-opa-30 colors/neutral-95-opa-0)
+              ]
```
Comment (Parveshdhull) [inline src/status_im/contexts/chat/composer/gradients/style.cljs]: Hi @clauxx, Thank you very much for your PR.
1. Please can you elaborate, how these changes are related to current issue?
2. Also, Sorry I am having little hard time locating these values. Please can you provide an link?

Comment (clauxx) [reply src/status_im/contexts/chat/composer/gradients/style.cljs]: 1. It's an additional UI fix to the composer, where for multiline messages there should be a bottom shadow at the bottom when collapsed/minimized. I mentioned it to QA in a comment above, but will add it to the issue description as well.
2. If you mean a design link, then the composer design should do it. But I didn't change the colors here, just adjusted the opacity and the gradient position so it looks according to the design.

**Comment by clauxx** [comment]
@Parveshdhull since you're already working on refactoring the composer and I don't have many changes here, we could close this given it'll be fixed in the refactor.

**Comment by Parveshdhull** [comment]
> @Parveshdhull since you're already working on refactoring the composer and I don't have many changes here, we could close this given it'll be fixed in the refactor.

No, it's alright. Refactoring of composer is gradual and probably will not be complete (and address this issue) before release. Please feel free to continue working on this, currently, it doesn't create any conflicts.

File: src/status_im/contexts/chat/composer/gradients/style.cljs (hunk 1 above)
Comment (Parveshdhull) [reply src/status_im/contexts/chat/composer/gradients/style.cljs]: > so it looks according to the design

Yes, that's what I am curious about. Please can you provide any reference what we are fixing here and what is not matching with design (maybe a screenshot or video). So we can know why it doesn't look design. Is it implementation issue or we are using different values then figma.

Comment (clauxx) [reply src/status_im/contexts/chat/composer/gradients/style.cljs]: Checked it against the designs and seems like this change is not necessary. I remember comparing it with the design at the time and the shadow being barely visible but it looks fine without these changes. Just the z-index fix should be enough to fix the bottom shadow, as before it wasn't visible at all. Thanks for being thorough with this :+1:

**Comment by clauxx** [comment]
Hi @mariia-skrypnyk! I checked [the issue](https://github.com/status-im/status-mobile/pull/18089#issuecomment-1872092842) that @churik mentioned, but can't reproduce it. Could you check if the issue is happening on your side?

**Comment by flexsurfer** [comment]
cc @pavloburykh

**Comment by pavloburykh** [comment]
@clauxx please rebase the branch and resolve the conflicts. After that we will continue testing.

**Comment by mariia-skrypnyk** [comment]
> Hi @mariia-skrypnyk! I checked [the issue](https://github.com/status-im/status-mobile/pull/18089#issuecomment-1872092842) that @churik mentioned, but can't reproduce it. Could you check if the issue is happening on your side?

Hi @clauxx !

Thanks for the update!
I have iPhone 13 iOS 17.2.1 and I am still catching composer jump.
Pavlo has iPhone X, IOS 16.7.2 and catches it also.

Let me show you my steps:

1. Open composer
2. Put text with short rows like I do on the video
3. You should have a lot of text to make cursor be at the bottom of composer
4. Tap on Return 10-20 times (at this step my cursor start hiding)
5. Try to type short text rows again

I attached my and Pavlo's videos:

https://github.com/status-im/status-mobile/assets/147824399/040140da-daa8-4d9d-9a65-635bfea30afc


https://github.com/status-im/status-mobile/assets/147824399/bb3ce326-937b-4eeb-b383-bb0164079bb6

Please, let me know if you need any additional help to catch this tricky issue!

**Comment by clauxx** [comment]
@mariia-skrypnyk Thanks for finding this! I think this issue is a bit different than the one reported by churik and the one I'm fixing here. Maybe would be best to have it as a separate issue, what do you think?

**Comment by mariia-skrypnyk** [comment]
> @mariia-skrypnyk Thanks for finding this! I think this issue is a bit different than the one reported by churik and the one I'm fixing here. Maybe would be best to have it as a separate issue, what do you think?



> @mariia-skrypnyk Thanks for finding this! I think this issue is a bit different than the one reported by churik and the one I'm fixing here. Maybe would be best to have it as a separate issue, what do you think?

No problem @clauxx. I add it as a new one:)
So, go ahead and merge this PR.

**Pull request merged**